sdk/eventgrid/Azure.Messaging.EventGrid/tests/EventGridClientLiveTests.cs

```diff
@@ -62,33 +62,69 @@ public async Task CanPublishEventToDomain()
         [Test]
         public async Task CanPublishCloudEvent()
         {
-            EventGridClientOptions options = Recording.InstrumentClientOptions(new EventGridClientOptions());
-            EventGridClient client = InstrumentClient(
-                new EventGridClient(
+            EventGridPublisherClientOptions options = Recording.InstrumentClientOptions(new EventGridPublisherClientOptions());
+            EventGridPublisherClient client = InstrumentClient(
+                new EventGridPublisherClient(
                     new Uri(TestEnvironment.CloudEventTopicHost),
                     new AzureKeyCredential(TestEnvironment.CloudEventTopicKey),
                     options));
             await client.PublishCloudEventsAsync(GetCloudEventsList());
         }
 
+        [Test]
+        public async Task CanPublishCloudEventWithBinaryData()
```

Comment:
We should try sending all of the different forms of binary data:
IEnumerable<byte>, byte[] (this is covered by IEnumerable<byte> but it would be good to test for it explicitly as well), ReadOnlyMemory<byte>, ReadOnlySpan<byte>